### PR TITLE
yuzu: update, switch to qt6, clean up, etc

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

Testing appreciated. Vulkan updates are in staging-next.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by K900** [comment]
staging-next merged, this should be a lot easier to test now.

**Comment by K900** [comment]
It should pick up qtwayland ambiently if you're running it in a Wayland session, though it might be worth adding to the wrapper as well.

**Comment by K900** [comment]
Let's just add it to the wrapper then.

**Comment by K900** [comment]
Also, tested a bit more - on 680M (Rembrandt iGPU), 6700S (Navi 23 mobile) and 6800XT (Navi 21 desktop), seems fine for me in both modes, no weird crashing, no weird colors on Gamescope or Plasma.

**Comment by K900** [comment]
Updated again since a bunch of perf fixes got merged to mainline. We should really set up an auto-updating flake for this...

**Review by athre0z — APPROVED** [review]
Works great here (NixOS, x86_64-linux). Thanks for the update!

**Comment by dmayle** [comment]
Trying to install here, if I don't manage, I'll test it once it's merged

**Comment by dmayle** [comment]
Tested as working over here, thanks for this!

**Comment by K900** [comment]
So, should I do another update and then merge this?

**Comment by K900** [comment]
Updated to latest versions, supposedly there's some more optimizations.

**Comment by baduhai** [comment]
Woking beautifully for me, the updates brought some great optimisations to the game I want to play :wink:.

File: pkgs/applications/emulators/yuzu/generic.nix [hunk 1]
```diff
@@ -57,71 +64,82 @@ stdenv.mkDerivation rec {
...
-    "-DYUZU_USE_EXTERNAL_SDL2=OFF"
+    # actually has a noticeable performance impact
+    "-DYUZU_ENABLE_LTO=ON"
 
+    # build with qt6
+    "-DENABLE_QT6=ON"
     "-DENABLE_QT_TRANSLATION=ON"
+
+    # use system libraries
+    "-DYUZU_USE_EXTERNAL_SDL2=OFF"
```
Comment (lheckemann) [inline pkgs/applications/emulators/yuzu/generic.nix:116]: Shouldn't this be ON if we want to use the system libraries?

**Review by lheckemann — APPROVED** [review]

**Pull request merged**